src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2560,7 +2571,7 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
                 HWIntrinsicImmOpHelper helper(this, intrin.op3, node);
 
-                for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
+                for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd(), (targetReg != op1Reg) ? 2 : 1)
```

Comment:
What is this syntax doing? It looks like the result of the ternary operator isn't being used.

Reply:
That change should have been on the line above - the `HWIntrinsicImmOpHelper` constructor needs to take in the number of instructions that will be generated in each stage of the loop.